src/network/networkd-link.c

```diff
@@ -1511,6 +1511,15 @@ static int link_up(Link *link) {
         }
 
         if (link->network->mtu) {
+                /* IPv6 protocol requires a minimum MTU of IPV6_MTU_MIN(1280) bytes
+                   on the interface. Bump up MTU bytes to IPV6_MTU_MIN. */
+                if (socket_ipv6_is_supported() && link->network->mtu < IPV6_MIN_MTU) {
```

Comment:
Yeah, I figure systemd should expose per-interface IPv4+IPv6 kill switches (or maybe derive this implicitly from whether SLAAC+DHCPv6 are both off, and no ipv6 address configured)